### ISSUE TITLE
Extending verify with user semantics

### ISSUE THREAD

**Issue by alexy** [scallop/scallop#4]
I often need to verify that if one option is set, then another is also set or unset, e.g. https://gist.github.com/2345102.  Perhaps this can be a part of verify?

**Comment by Rogach** [comment]
I'm thinking of something like this:

``` scala
object Conf extends ScallopConf(arguments) {
  val date = opt[String]("date")
  val datetime = opt[String]("datetime")
  mutuallyExclusive("date", "datetime") // takes varargs
  verify 
}
```

Also there would be needed some other kinds of relations (mutuallyInclusive :) ).

What do you think?

**Comment by alexy** [comment]
Definitely makes sense as it's a constraint in the option world which everybody can impose.

**Comment by Rogach** [comment]
Ok, now I just need a better name for that "mutuallyInclusive" :)

Also I planned to add per-option validatiors - something like

``` scala
val size = opt[Int]("size", validate = (0<))
```

Will those be handy?

**Comment by alexy** [comment]
Most definitely, single-option validation is the most typical one.  mutuallyInclusive=>ensemble!  Or codependent...

**Comment by Rogach** [comment]
I like codependent more :)

**Comment by Rogach** [comment]
I implemented this. Now, you can call the .mutuallyExclusive and .codependent methods with option names:

``` scala
object Conf extends ScallopConf(List("-a","1","-b","2")){
  val apples = opt[Int]("apples")
  val bananas = opt[Int]("bananas")
  codependent(apples, bananas)
  verify
}
```

Also, I implemented a method for custom cross-option validation logic:

``` scala
object Conf extends ScallopConf(List("-a","14","-b","3")) {
  val apples = opt[Int]("apples")
  val bananas = opt[Int]("bananas")
  validate (apples, bananas) { (a,b) =>
    if (a > 0 && b % 7 == 0) Right(Unit)
    else Left("Something is wrong with composition :)")
  }
  verify
}
```

This method automatically defined co-dependent relationship on all needed options. `.validate` methods are provided up to the arity of 7. 

All of this is already working in `master` and would be published in 0.3.5.
